src/sql/parts/modelComponents/declarativeTable.component.ts

```diff
@@ -0,0 +1,194 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import 'vs/css!./declarativeTable';
+import {
+	Component, Input, Inject, ChangeDetectorRef, forwardRef, ComponentFactoryResolver,
+	ViewChild, ViewChildren, ElementRef, Injector, OnDestroy, QueryList, AfterViewInit
+} from '@angular/core';
+
+import * as sqlops from 'sqlops';
+
+import { ComponentBase } from 'sql/parts/modelComponents/componentBase';
+import { IComponent, IComponentDescriptor, IModelStore, ComponentEventType } from 'sql/parts/modelComponents/interfaces';
+import { IWorkbenchThemeService } from 'vs/workbench/services/themes/common/workbenchThemeService';
+import { IContextViewService } from 'vs/platform/contextview/browser/contextView';
+import { Event, Emitter } from 'vs/base/common/event';
+import { Checkbox } from 'sql/base/browser/ui/checkbox/checkbox.component';
+import { SelectBox } from 'sql/base/browser/ui/selectBox/selectBox.component';
+import { ISelectData } from 'vs/base/browser/ui/selectBox/selectBox';
+import { InputBox } from 'sql/base/browser/ui/inputBox/inputBox.component';
+import * as nls from 'vs/nls';
+
+export enum DeclarativeDataType {
+	string = 'string',
+	category = 'category',
+	boolean = 'boolean'
+}
+
+@Component({
+	selector: 'modelview-declarativeTable',
+	template: `
+	<table role=grid aria-labelledby="ID_REF" #container *ngIf="columns" class="declarative-table">
+		<ng-container *ngFor="let column of columns;let h = index">
+			<th class="declarative-table-header" tabindex="-1" role="button" aria-sort="none">{{column.displayName}}</th>
```

Comment:
>role="button" [](start = 54, length = 13)

Header's role is button? Is it right?

Reply:
I work on accessibility in another PR

---
In reply to: [194470319](https://github.com/Microsoft/sqlopsstudio/pull/1593#discussion_r194470319) [](ancestors = 194470319)